larpandoracontent/LArObjects/LArCaloHit.h

```diff
@@ -82,8 +90,10 @@ class LArCaloHit : public object_creation::CaloHit::Object
 
 private:
     unsigned int            m_larTPCVolumeId;       ///< The lar tpc volume id
+    unsigned int            m_daughterVolumeId;     ///< The daughter volume id
```

Comment:
Probably makes sense to do it this way. Just putting it out there, that this could be a pandora::InputType (pandora::InputUInt, exists, I think?), so that it could be explicitly uninitialised if the information isn't there.

The advantage of that is that any code accessing this volume id will bail, if it wasn't provided.

The disadvantage is that it's more complicated, there's a small overhead, and we might not want everything to bail, but just proceed with the view that everything has the same daughter volume id.

So, probably fine, but just putting this down for the record.

Reply:
I think this approach is consistent with the handling of m_larTPCVolumeId? i.e. the equivalent variable in LArCaloHitParameters is defined as an InputUInt, and then here it is a primitive uint.

Reply:
Yes, I think it's good that it's treated in the same way as the main "lar tpc" volume id. The only difference I can think of is that the reconstruction strategy now mandates that all hits have a volume id (otherwise you can't work in the context of the master algorithm, and need to go back to processing all input hits in one go). The daughter volume id, meanwhile, has no algs using it yet, and the strategy might be just to have one or two algs that look at the id. So optional is currently possible, but consistency with an existing id is a strong motivation, and probably wins.

Reply:
Yes, this is done in the same way as for m_larTPCVolumeId. Do we want to change the behaviour for both, or leave as is?

Reply:
Hi, I don't think we should change the implementation of the mandatory volume id, so the question is how we handle the optional daughter volume id. It's looking like it's going to remain optional from the app (or input pndr/xml) side. We could keep it optional in the LArCaloHits too, but I think we're leaning towards this implementation, where it's a simple unsigned int member variable, defaulted to zero.

The only thing we lose by doing that is knowledge of when our algs are using/accessing the daughter volume id when one hasn't been explicitly provided. Moving to a pandora::InputUInt would make the algs bail when information that wasn't explicitly provided was accessed. All the other input parameters (barring the provision of track-like and shower-like probabilities) _must_ be explicitly provided by the app/pndr/xml, I think.

It's just this new one that the app/pndr/xml can skip providing, but we'll (in LArContent) choose what it's default value shall be. Hence it's worth having this discussion. Shall we say the unsigned int (default value zero) is a preferred option right now (as opposed to pandora::InputUInt)? Thanks!

Reply:
Thanks for the clarifications @PandoraPFA  - I actually hadn't seen your previous comment when I submitted mine... so, sorry for the repetition!

At the moment we use this daughter volume ID in matching algorithms in [this PR down the line](https://github.com/PandoraPFA/LArContent/pull/142) in a check on whether cluster hits share a common daughter volume, so there should be no issues if the daughter volume ID is a default value, same for all.

So yes, happy to leave it as it is for now! Thanks!